src/test/java/org/jabref/logic/util/io/FileNameHandlerTest.java

```diff
@@ -0,0 +1,54 @@
+package org.jabref.logic.util.io;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class FileNameHandlerTest {
+
+    @Test
+    public void testCleanFileName() {
+        assertEquals("legalFilename.txt", FileNameHandler.cleanFileName("legalFilename.txt"));
+        assertEquals("illegalFilename______.txt", FileNameHandler.cleanFileName("illegalFilename/?*<>|.txt"));
+    }
+
+    @Test
+    public void testCleanDirectoryName() {
+        assertEquals("legalFilename.txt", FileNameHandler.cleanDirectoryName("legalFilename.txt"));
+        assertEquals("subdir/legalFilename.txt", FileNameHandler.cleanDirectoryName("subdir/legalFilename.txt"));
+        assertEquals("illegalFilename/_____.txt", FileNameHandler.cleanDirectoryName("illegalFilename/?*<>|.txt"));
+    }
+
+    @Test
+    public void testCleanDirectoryNameForWindows() {
+        assertEquals("legalFilename.txt", FileNameHandler.cleanDirectoryName("legalFilename.txt"));
+        assertEquals("subdir\\legalFilename.txt", FileNameHandler.cleanDirectoryName("subdir\\legalFilename.txt"));
+        assertEquals("illegalFilename\\_____.txt", FileNameHandler.cleanDirectoryName("illegalFilename\\?*<>|.txt"));
+    }
+
+    @Test
+    public void testCleanCurlyBracesAsWell() {
+        assertEquals("The Evolution of Sentiment_ Analysis_A Review of Research Topics, Venues, and Top Cited Papers.PDF", FileNameHandler.cleanFileName("The Evolution of Sentiment} Analysis}A Review of Research Topics, Venues, and Top Cited Papers.PDF"));
+    }
+
+    @Test
+    public void testGetNonOverWritingFileNameReturnsUniqueName(@TempDir Path tempDirectory) throws IOException {
```

Comment:
Good to have a test. - Could you add two more:

1. One for a non existing file
2. One for just one file existing

I know, these are trivial. - Nevertheless, if someone touches your code, even the trivial ones are tested then (and there could something go wrong)